src/App.tsx

```diff
@@ -1,10 +1,43 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from "react";
+import { UserWarning } from "./UserWarning";
+import { getTodos } from "./api/todos";
+import { Todo } from "./types/Todo";
+import { ListofTodo } from "./Components/ListofTodo/ListofTodo";
+import { Error } from "./Components/Error/Error";
+import { FilterTodo } from "./Components/FilterTodo/FilterTodo";
+import { FilterStatus } from "./types/FilterStatus";
 
-const USER_ID = 0;
+const USER_ID = 6429;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState("");
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.all);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const unfinishedTodos = todos.filter((todo) => !todo.completed);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((result) => setTodos(result))
+      .catch(() => setError("Unable to load the todos"));
+  }, []);
+
+  useEffect(() => {
+    const currentTodos = todos.filter((todo) => {
+      switch (filter) {
+        case FilterStatus.active:
+          return !todo.completed;
+
+        case FilterStatus.completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+
+    setFilteredTodos(currentTodos);
```

Comment:
You don't really need an additional state for filtered todos. Just wrap currentTodos in useMemo and that's it